docs/bridge/developer-guide-vara-eth.mdx

```diff
@@ -0,0 +1,323 @@
+---
+title: Vara ↔ Ethereum App guilde
+sidebar_position: 5
+sidebar_label: Vara ↔ Eth App guide
+---
+
+# Introduction
+
+Cross-chain communication can seem complex, but at its core, it’s about securely moving messages and value between independent blockchains. To make this process clear and approachable, we’ll build a minimal “ping-pong” application on top of the **Vara ↔ Ethereum Bridge**.
+
+:::note
+The ping-pong pattern is a classic example used in distributed systems to demonstrate reliable message delivery: a simple “ping” sent from one chain and a “pong” received in response from the other.
+:::
+
+In this first part, we focus on the flow from Vara to Ethereum using the bridge’s built-in ZK-based mechanism. We’ll walk through how the protocol handles the heavy lifting—cryptographic proofs, relaying, and validation—while letting developers concentrate on the application logic. In future articles, we’ll explore the reverse direction (Ethereum to Vara) and additional cross-chain patterns.
+
+## High-Level Architecture
+
+At a high level, the cross-chain ping-pong app consists of three main parts, each focused on its own responsibility while relying on the bridge’s secure protocol layer:
+
+1. **On Vara:** a Gear Program on Rust acts as the Ping sender. It packages a simple payload and sends it using the built-in bridge API, specifying the destination contract on Ethereum.  
+   [See Vara Program Code](https://github.com/gear-foundation/vara-eth-cross-ping/tree/main/cross-ping)
+2. **In the middle:** a permissionless relayer listens for events, retrieves the required Merkle root (which guarantees the message’s existence in the queue), obtains the necessary 'Merkle proof', and—at the right moment, when Ethereum’s state is updated—submits 'the message', 'root', and 'proof' to Ethereum.  
+   [See Relayer Code](https://github.com/gear-foundation/vara-eth-cross-ping/tree/main/relayer)
```

Comment:
In the Ralyer readme, mention that .env is just an example, the actual bridge addresses are in [wiki link].